File: server/router/hospitalManage/jobTitle.go
package hospitalManage

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type JobTitle struct{}

// 职称管理
func (j *JobTitle) InitJobTitleRouter(Router *gin.RouterGroup, RouterPub *gin.RouterGroup) {
	jobTitleRouter := Router.Group("jobTitle").Use(middleware.OperationRecord())
	JobTitleApi := v1.ApiGroupApp.HospitalManageApiGroup.JobTitleApi
	{
		jobTitleRouter.GET("getList", JobTitleApi.GetJobTitleList)
		jobTitleRouter.POST("create", JobTitleApi.CreateJobTitle)
		jobTitleRouter.POST("update", JobTitleApi.UpdateJobTitle)
		jobTitleRouter.DELETE("delete", JobTitleApi.DeleteJobTitle)
	}
}

File: web/src/api/jobTitle.js
import service from "@/utils/request";

export const getList = (params) => {
  return service({
    url: "/jobTitle/getList",
    method: "get",
    params,
  });
};
export const createJobTitle = (data) => {
  return service({
    url: "/jobTitle/create",
    method: "post",
    data,
  });
};
export const updateJobTitle = (data) => {
  return service({
    url: "/jobTitle/update",
    method: "post",
    data,
  });
};
export const deleteJobTitle = (params) => {
  return service({
    url: "/jobTitle/delete",
    method: "delete",
    params,
  });
};
